Pa3/Durchführung/Project/app/StateHandler.c

```diff
@@ -31,13 +31,190 @@
 /* PROTOTYPES *************************************************************************************/
 
 /* VARIABLES **************************************************************************************/
-static EventEnum gCurrentState;
+static EventEnum gCurrentEvent = NO_EVENT_HAS_HAPPEND; 
+static StateEnum gCurrentState = STATE_INIZALIZATION_MCU; //First state when the MCU is started
 static SoftTimer* gLapTimer;
+static bool gEntryDone = false;
 
 /* EXTERNAL FUNCTIONS *****************************************************************************/
 
 void StateHandler_process(void)
 {
+  switch(gCurrentState)
+  { 
+    case STATE_INIZALIZATION_MCU: 
+      InitializeMcu_InitializeAll(void); //entry
+      InitializeMcu_DisplayTeamName(void); //exit
+      gCurrentState = STATE_CALIBRATE_LINE_SENSORS;
+      break;
+
+    case STATE_CALIBRATE_LINE_SENSORS:
+      if(gEntryDone == false)
+      {
+        gEntryDone = true;
+        gCurrentEvent = CalibrateLineSensors_Initialize(void); //entry
+        if (NO_EVENT_HAS_HAPPEND != gCurrentEvent)
+        {
+          gCurrentState = STATE_ERROR_HANDLER;
+        }
+      }
+
+      gCurrentEvent = CalibrateLineSensors_CalibrateSensors(void); //do
+      switch (gCurrentEvent)
+      {
+        case CALIBRATION_FAILED:
+          gCurrentState = STATE_ERROR_HANDLER;
```

Comment:
why is entryDone not reset on Error-transition? the ErrorState checks this flag to determine it's own entry status